kb/2023-07-25/configure_priority_class_longhorn.md

```diff
@@ -0,0 +1,314 @@
+---
+title: Configure PriorityClass on Longhorn System Components
+description: Configure priority classes on Longhorn system components
+slug: configure_priority_class_longhorn
+authors:
+  - name: Kiefer Chang
+    title: Engineer Manager
+    url: https://github.com/bk201
+    image_url: https://github.com/bk201.png
+tags: [harvester, longhorn, "priority class"]
+hide_table_of_contents: false
+---
+
+Starting from Harvester `v1.2.0`, Longhorn system-managed components in new-deployed clusters have `PriorityClass` set to `system-cluster-critical` by default. But Longhorn system-managed components won't have any `PriorityClass` if a user upgrades a Harvester cluster from previous versions. 
+
+This phenomenon is because we want to support zero-downtime upgrades, but Longhorn doesn't allow changing the `priority-class` setting when there are attached volumes (Please see the [doc](https://longhorn.io/docs/1.4.3/advanced-resources/deploy/priority-class/#setting-priority-class-during-longhorn-installation)).
+
+This article describes manually configuring priority classes for Longhorn system-managed components after an upgrade.
+
+## Stop all virtual machines
+
+You must stop all virtual machines to detach all volumes. Please back up any work before doing this.
+- [Log in to a Harvester controller node and become root](https://docs.harvesterhci.io/v1.1/troubleshooting/os#how-to-log-into-a-harvester-node).
+- Get all running virtual machines and write down their namespaces and names:
```

Comment:
do we need to mentioned that users can also do this via the UI dashboard?

Reply:
Sure, I add a note in fb8a5558f6e89a8ae9ca5f7485ad8b18c7793d81.
Also, I encourage users to shut down inside the VMs. The Stop button has a timeout and can kill VM after it.